src/main.rs

```diff
@@ -1311,78 +1311,95 @@ fn parse_manpage_at_path(
-//             # Pad on the right with spaces so we overwrite whatever we wrote last time
-//             padded_progress_str = progress_str.ljust(last_progress_string_length)
-//             last_progress_string_length = len(progress_str)
-//             sys.stdout.write("\r{0}\r".format(padded_progress_str))
-//             sys.stdout.flush()
-//
-//         # Maybe we want to skip this item
-//         skip = False
-//         if not WRITE_TO_STDOUT:
-//             # Compute the path that we would write to
-//             output_path = os.path.join(output_directory, output_file_name)
-//
-//         # Now skip if requested
-//         if skip:
-//             continue
-//
-//         try:
-//             if parse_manpage_at_path(manpage_path, output_directory):
-//                 successful_count += 1
-//         except IOError:
-//             diagnostic_indent = 0
-//             add_diagnostic('Cannot open ' + manpage_path)
-//         except (KeyboardInterrupt, SystemExit):
-//             raise
-//         except:
-//             add_diagnostic('Error parsing %s: %s' % (manpage_path, sys.exc_info()[0]), BRIEF_VERBOSE)
-//             flush_diagnostics(sys.stderr)
-//             traceback.print_exc(file=sys.stderr)
-//         flush_diagnostics(sys.stderr)
-//     print("") #Newline after loop
-//     add_diagnostic("Successfully parsed %d / %d pages" % (successful_count, total_count), BRIEF_VERBOSE)
-//     flush_diagnostics(sys.stderr)
-
-#[derive(Copy, Clone, Debug)]
-struct Progress(pub bool);
-
-// TODO Arg/output types?
+/// Get the number of digits in num
+fn num_digits(n: usize) -> usize {
+    (1.max(n) as f32).log10() as usize + 1
+}
+
+#[test]
+fn test_num_digits() {
+    assert_eq!(num_digits(1000), 4);
+    assert_eq!(num_digits(100), 3);
+    assert_eq!(num_digits(33), 2);
+    assert_eq!(num_digits(123456789012345), 15);
+    assert_eq!(num_digits(0), 1);
+}
+
 fn parse_and_output_man_pages(
-    _paths: impl Iterator<Item = PathBuf>,
-    _output_directory: PathBuf,
-    Progress(_show_progress): Progress,
+    paths: &mut [PathBuf],
+    output_directory: PathBuf,
+    show_progress: bool,
+    deroff_only: bool,
+    write_to_stdout: bool,
 ) {
-    unimplemented!();
+    paths.sort();
+
+    let total = paths.len();
+
+    let mut successful_count = 0;
+    let padding = num_digits(total);
+
+    let mut last_len: usize = 0;
+
+    if show_progress && !write_to_stdout {
+        println!(
+            "Parsing man pages and writing completions to {:?}",
+            output_directory
+        );
+    }
+
+    for (index, manpage_path) in paths.iter().enumerate() {
+        // foo/bar/gcc.1.gz -> gcc.1.gz
+        let man_file_name = manpage_path
+            .file_name()
+            .map(|fname| fname.to_string_lossy())
+            .unwrap_or_else(|| panic!("Failed to get manfile name from {:?}", manpage_path));
```

Comment:
```suggestion
            .unwrap_or_else(|| panic!("Failed to get manfile name from {}", manpage_path.as_ref()));
```

Reply:
These don't work, it can't tell what you're trying to `as_ref` as, also, i dont know why as_ref() is better in this context?

Reply:
Because it will be printed out as string.

Reply:
Do you mean like `"path/to/whatever.zip"` -> `path/to/whatever.zip`?
if so, we'd need to do `manpage_path.as_path().display()`, because it can't tell if you're trying to `as_ref()` into an `OsStr` or a `Path`

Reply:
IIRC `as_ref()` will turn it into something that is `Display`, I don't know if it's an `OsStr` or `Path` too, I just know that it becomes something displayable.

Reply:
PathBuf is `Deref<Target=Path>`, `Path` has a [`display`](https://doc.rust-lang.org/std/path/struct.Path.html#method.display) method that makes it `Display`, I'll add this as `manpage_path.display()`

Reply:
5cae975